tests/46direct/01directmessage.pl

```diff
@@ -70,7 +70,9 @@ sub matrix_recv_device_message
    } until => sub {
       my ( $f ) = @_;
       return 1 if $f->failure;
-      scalar @{ $f->get->{to_device}{events} };
+      my $resp = $f->get;
+      log_if_fail "Sync response", $resp;
+      scalar @{ $resp->{to_device}{events} };
```

Comment:
I know `scalar` gets the length of an array in perl, but why are we doing that here without storing the value at all?

Or does the value here get returned?

Probably touches on the point that sytest could use more comments.

Reply:
yeah, in perl, the result of a block is the result of the last expression in that block, so there's an implicit `return` here.

> Probably touches on the point that sytest could use more comments.

possibly, although I think it's better to try to write code in a way that's clear. In this case I think an explicit `return` is preferable. In general I think `sytest` relies too much on implicit `return`.